pinot-common/src/main/java/org/apache/pinot/common/utils/helix/LeadControllerUtils.java

```diff
@@ -58,4 +69,42 @@ public static String generatePartitionName(int partitionId) {
   public static int extractPartitionId(String partitionName) {
     return Integer.parseInt(partitionName.substring(partitionName.lastIndexOf('_') + 1));
   }
+
+  /**
+   * Checks from ZK if resource config of leadControllerResource is enabled.
+   * @param helixManager helix manager
+   */
+  public static boolean isLeadControllerResourceEnabled(HelixManager helixManager) {
+    try {
+      HelixDataAccessor helixDataAccessor = helixManager.getHelixDataAccessor();
+      PropertyKey propertyKey = helixDataAccessor.keyBuilder().resourceConfig(Helix.LEAD_CONTROLLER_RESOURCE_NAME);
+      ResourceConfig resourceConfig = helixDataAccessor.getProperty(propertyKey);
+      String resourceEnabled = resourceConfig.getSimpleConfig(Helix.LEAD_CONTROLLER_RESOURCE_ENABLED_KEY);
+      return Boolean.parseBoolean(resourceEnabled);
+    } catch (Exception e) {
+      LOGGER.warn("Could not get whether lead controller resource is enabled or not.", e);
```

Comment:
Why are we not throwing exception here? 
At the minimum, we need to indicate in the API that a 'false' return does not mean much, but I prefer we throw exception and let the caller handle it.